ui-test/src/main/java/pages/SignupFormDynamicFiller.java

```diff
@@ -0,0 +1,127 @@
+	private WebDriver driver;
+	private static final Logger logger = Logger.getLogger(SignupFormDynamicFiller.class);
+	RegistrationPage registrationPage;
+
+	public SignupFormDynamicFiller(WebDriver driver) {
+		this.driver = driver;
+		registrationPage = new RegistrationPage(driver);
+	}
+
+	public void fillFormFromUiSpec(Map<String, Map<String, Object>> uiSpecFields) {
+
+		for (String fieldId : uiSpecFields.keySet()) {
+
+			if (fieldId.equalsIgnoreCase("phone")) {
+				continue;
+			}
+
+			List<WebElement> matchingElements = driver
+					.findElements(By.xpath("//*[@id='" + fieldId + "' or @data-field-id='" + fieldId + "']"));
+
+			if (matchingElements.isEmpty()) {
+				logger.info("No element found for fieldId: " + fieldId);
+				continue;
+			}
+
+			WebElement element = matchingElements.get(0);
+			String tag = element.getTagName();
+			String type = element.getAttribute("type");
+
+			if (fieldId.equalsIgnoreCase("individualBiometrics")) {
+				registrationPage.clickOnUploadPhoto();
+				registrationPage.clickOnCaptureButton();
+				continue;
+			}
+
+			if ("checkbox".equalsIgnoreCase(type) || fieldId.equalsIgnoreCase("consent")) {
+				if (!element.isSelected()) {
+					((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView({block: 'center'});",
+							element);
+					((JavascriptExecutor) driver).executeScript("arguments[0].click();", element);
+				}
+				continue;
+			}
+
+			if ("hidden".equalsIgnoreCase(type) || !element.isEnabled()) {
+				continue;
+			}
+
+			if (fieldId.toLowerCase().contains("name")) {
+				EsignetUtil.FullName names = EsignetUtil.generateNamesFromUiSpec();
+
+				for (WebElement nameField : matchingElements) {
+					String lang = nameField.getAttribute("data-lang");
+
+					if ("eng".equalsIgnoreCase(lang)) {
+						nameField.clear();
+						nameField.sendKeys(names.english);
+					} else if ("khm".equalsIgnoreCase(lang)) {
+						nameField.clear();
+						nameField.sendKeys(names.khmer);
+						RegisteredDetails.setFullName(names.khmer);
+					} else {
+						logger.info("Full name entry not supported for language: " + lang);
+					}
+				}
+				continue;
+			}
+
+			if (fieldId.equalsIgnoreCase("password")) {
+				String password = EsignetUtil.generateValidPasswordFromActuator();
+				RegisteredDetails.setPassword(password);
+				element.clear();
+				element.sendKeys(password);
+
+				WebElement confirmPwd = driver.findElement(By.id("password_confirm"));
+				confirmPwd.clear();
+				confirmPwd.sendKeys(password);
+				continue;
+			}
+
+			if (fieldId.equalsIgnoreCase("email")) {
+				String email = EsignetUtil.generateEmailFromRegex(fieldId);
+				element.clear();
+				element.sendKeys(email);
+				continue;
+			}
+
+			if (tag.equalsIgnoreCase("select")) {
+				Select dropdown = new Select(element);
+				List<WebElement> options = dropdown.getOptions();
+				if (options.size() > 1) {
+					int index = new Random().nextInt(options.size() - 1) + 1;
+					dropdown.selectByIndex(index);
```

Comment:
These conditions on fieldId is not correct. Can these check happen based on the controlType?